src/basal_mass_balance/BMB_main.f90

```diff
@@ -111,6 +127,16 @@ SUBROUTINE run_BMB_model( mesh, ice, ocean, refgeo, SMB, BMB, region_name, time)
         CALL crash('unknown choice_BMB_model "' // TRIM( choice_BMB_model) // '"')
     END SELECT
 
+    ! Apply subgrid scheme of old BMB to new mask
+    SELECT CASE (choice_BMB_model)
+      CASE ('inverted')
+        ! No need to do anything
+      CASE ('prescribed_fixed')
+        ! No need to do anything
+      CASE DEFAULT
+        CALL apply_BMB_subgrid_scheme( mesh, ice, BMB)
+    END SELECT
```

Comment:
At some point we're going to figure out a way to prevent this code duplication...

Reply:
Sounds good, I'm sure some things can be made more compact here and there. Anyway there's still some aspects of the subgrid scheme we need to discuss. I'll leave this as is for now and open an issue to clean this stuff up